pkg/chains/formats/intotoite6/taskrun/material.go

```diff
@@ -0,0 +1,205 @@
+/*
+Copyright 2022 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package taskrun
+
+import (
+	"encoding/json"
+	"strings"
+
+	slsa "github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/v0.2"
+	"github.com/tektoncd/chains/pkg/artifacts"
+	"github.com/tektoncd/chains/pkg/chains/formats/intotoite6/attest"
+	"github.com/tektoncd/chains/pkg/chains/objects"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"github.com/tektoncd/pipeline/pkg/apis/resource/v1alpha1"
+	"go.uber.org/zap"
+)
+
+const (
+	uriSeparator    = "@"
+	digestSeparator = ":"
+)
+
+// RemoveDuplicateMaterials removes duplicate materials from the slice of materials
+func RemoveDuplicateMaterials(mats []slsa.ProvenanceMaterial) ([]slsa.ProvenanceMaterial, error) {
+	// make materialMap to store unique materials
+	materialMap := map[string]bool{}
```

Comment:
nit:
To create a set
map[string]struct{} {}
is better.
struct{} takes no space and its value can never be confused with the exists value.

Reply:
great! thanks!